### PR TITLE
[Vulkan] Add support of loading AOT modules and fields

### PR DESCRIPTION
**Pull request by ghuau-innopeak**
Related issue = #3679

### CODE REVIEW

**Comment by ghuau-innopeak** [comment]
For `to_vk_dtype_enum`, right now, I'm using constant values with a static function, but I'm wondering if we even need to serialize dtype as we already serialize the name?

File: taichi/backends/vulkan/aot_module_builder_impl.cpp [hunk 1]
```diff
@@ -46,6 +86,26 @@ void AotModuleBuilderImpl::dump(const std::string &output_dir,
   ts.write_to_file(txt_path);
 }
 
+void AotModuleBuilderImpl::load(const std::string &output_dir,
+                                const std::string &filename) {
+  const std::string bin_path =
+      fmt::format("{}/{}_metadata.tcb", output_dir, filename);
```
Comment (ailzhang) [inline taichi/backends/vulkan/aot_module_builder_impl.cpp:92]: nit: filename is no longer used? ;)

**Review by ailzhang — APPROVED** [review]
Looks great! Thanks! 
Shall we add a C++ test saving & loading as followup? ;)

File: taichi/backends/vulkan/aot_module_builder_impl.cpp (hunk 1 above)
Comment (ghuau-innopeak) [reply taichi/backends/vulkan/aot_module_builder_impl.cpp]: Oops, my bad! should be fixed :)

**Comment by ghuau-innopeak** [comment]
> Looks great! Thanks! Shall we add a C++ test saving & loading as followup? ;)

@ailzhang Indeed, and some example code in C++ for `aot_save` was merged recently that could be re-used for testing too (#3707)

**Pull request merged**